### PR TITLE
Blog Post: Understanding and Controlling AWS Lambda Costs

### PR DESCRIPTION
**Pull request by mayanklahiri**
Hi folks,

This is an article I've been talking to @rupakg about. Had some issues with the parser not accepting a Medium image URL (trailing period in URL seems to be omitted), hope it's not too much of a fix.

### CODE REVIEW

File: posts/2017-09-14-understanding-and-controlling-aws-lambda-costs.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+
+* **Processing delays can be expensive**. Suppose that a 512 MB Lambda function uses executes in slightly less than **200 milliseconds**, with a hard limit of 5 seconds. As part of its processing, the function calls an external service over HTTPS and waits for a response before ending. Suppose that network congestion or an external service degradation adds a spike of **2 seconds** to each network call. For the duration of the latency spike, **the extra 2 seconds of Lambda running time will increase cost *11 times***, from \$0.17 per 100k requests to \$1.83 ([spreadsheet](https://docs.google.com/spreadsheets/d/1xh_rNQwGVLos7Fshq2eM4gbIz546LwvrY1fMcLGqPmw/edit?zx=p7c6vgh0j97k#gid=541900098)).
+
```
Comment (rupakg) [inline posts/2017-09-14-understanding-and-controlling-aws-lambda-costs.md]: "Processing delays can be expensive. Suppose that a 512 MB Lambda function ~~uses~~ executes in slightly less ..."

**Comment by rupakg** [comment]
@mayanklahiri Very nice write up. I was curious if you could open source the code that you used to derive at the benchmarks. 

Also, can you give the reader an idea as to how you:
- capped the code/function to run within the desired execution time
- measure the execution time of the code/function
Maybe that will be a good test that can be done prior to deploying code to make sure to stay within the expected execution time boundaries.

I am planning on publishing this post tomorrow.

**Comment by mayanklahiri** [comment]
@rupakg Sounds good. I've added more details about measurement to the post body. The code and data is actually already all open (the full experimentation framework too). I've added more links in the relevant places.

**Comment by alexdebrie** [comment]
Great post, @mayanklahiri! Thanks for contributing.

@rupakg This is good on my end.

**Comment by rupakg** [comment]
@mayanklahiri Looks good.

**Comment by rupakg** [comment]
schedule(09/15/2017 6:00 am)

**Comment by DavidWells** [comment]
✅ Post scheduled for **09/15/2017 at 6:00am (America/Los_Angeles)**

**To unscheduled the post:**

1. **[Delete the previous comment ⬆](https://github.com/serverless/blog/pull/303#issuecomment-329696862)** with schedule(time) information by clicking the ❌
  ```
  ┌──────────────────────────────────────────────┐
  │                                           X  │
  ├───────────────────────────────────────────▲──┤
  │                                           │  │
  │  schedule(mm/dd/yyyy h:mm A)   ┌────────┐ │  │
  │                                │ Delete ├─┘  │
  │                                └────────┘    │
  └──────────────────────────────────────────────┘
  ```
2. (Optionally) Delete this comment to keep things tidy

> Powered by the ⚡️ [Serverless post scheduler](https://github.com/serverless/post-scheduler) | Made with ❤️  by [Serverless](https://serverless.com/)

**Comment by DavidWells** [comment]
🎉 **The post has been published**

> Powered by the ⚡️ [Serverless post scheduler](https://github.com/serverless/post-scheduler) | Made with ❤️  by [Serverless](https://serverless.com/)

**Pull request merged**